src/core/quadtree.js

```diff
@@ -0,0 +1,462 @@
+/*jslint vars: true, nomen: true, plusplus: true, continue:true, forin:true */
+/*global Node, QNode */
+
+/*
+
+  Copyright (c) 2014 Michael Sander (speedplane)
+  Based loosely on code written by (c) 2011 Mike Chambers.
```

Comment:
For reference: the licenses we are allowed to use, when including code in Firefox, are listed under the "Licenses Compatible with the MPL" heading in https://www.mozilla.org/MPL/license-policy.html.